features/environment.py

```diff
@@ -139,11 +187,41 @@ def from_environ(cls, config) -> "UAClientBehaveConfig":
 def before_all(context: Context) -> None:
     """behave will invoke this before anything else happens."""
     userdata = context.config.userdata
+    if userdata:
+        print("Userdata key / value pairs:")
+        for key, value in context.config.userdata.items():
+            print("   - {} = {}".format(key, value))
     context.series_image_name = {}
     context.series_reuse_image = ""
     context.reuse_container = {}
     context.config = UAClientBehaveConfig.from_environ(context.config)
-
+    has_aws_keys = bool(
+        context.config.aws_access_key_id
+        and context.config.aws_secret_access_key
+    )
+    if not has_aws_keys:
+        context.config.ec2_api = None
+    else:
+        logging.basicConfig(
+            filename="pycloudlib-behave.log", level=logging.DEBUG
+        )
+        context.config.ec2_api = pycloudlib.EC2(
+            tag="ua-testing",
+            access_key_id=context.config.aws_access_key_id,
+            secret_access_key=context.config.aws_secret_access_key,
+        )
+        ec2_api = context.config.ec2_api
+        if "uaclient-integration" not in ec2_api.list_keys():
+            keypair = ec2_api.client.create_key_pair(
+                KeyName="uaclient-integration"
+            )
+            with open("uaclient1.pem", "w") as stream:
+                stream.write(keypair["KeyMaterial"])
+            os.chmod("uaclient1.pem", 0o600)
+        ec2_api.use_key(
+            "uaclient1.pem", "uaclient1.pem", "uaclient-integration"
+        )
+        vpc = ec2_api.create_vpc("uaclient-integration")
```

Comment:
I don't know if I am missing something, but why do we need to create this vpc here ?

Reply:
ahh oversight from an earlier test. I've removed it now.